fdbserver/KeyValueStoreShardedRocksDB.actor.cpp

```diff
@@ -679,22 +679,29 @@ rocksdb::WALRecoveryMode getWalRecoveryMode() {
 
 rocksdb::ColumnFamilyOptions getCFOptions() {
 	rocksdb::ColumnFamilyOptions options;
-	options.memtable_max_range_deletions = SERVER_KNOBS->ROCKSDB_MEMTABLE_MAX_RANGE_DELETIONS;
-	options.disable_auto_compactions = SERVER_KNOBS->ROCKSDB_DISABLE_AUTO_COMPACTIONS;
-	options.level_compaction_dynamic_level_bytes = SERVER_KNOBS->ROCKSDB_LEVEL_COMPACTION_DYNAMIC_LEVEL_BYTES;
-	options.OptimizeLevelStyleCompaction(SERVER_KNOBS->ROCKSDB_MEMTABLE_BYTES);
+
+	if (SERVER_KNOBS->ROCKSDB_LEVEL_COMPACTION_DYNAMIC_LEVEL_BYTES) {
+		options.level_compaction_dynamic_level_bytes = SERVER_KNOBS->ROCKSDB_LEVEL_COMPACTION_DYNAMIC_LEVEL_BYTES;
+		options.OptimizeLevelStyleCompaction(SERVER_KNOBS->SHARDED_ROCKSDB_MEMTABLE_BUDGET);
```

Comment:
Does this have to be in this if loop?
ROCKSDB_LEVEL_COMPACTION_DYNAMIC_LEVEL_BYTES is randomized in simulation. If simulation picks false, this OptimizeLevelStyleCompaction is not called. The default compaction_style is levelStyle but we the optimization code is not called (infact not sure how much it is useful though, since we are changing the parameters)

Reply:
The parameters overrides are actually the same as the output generated from `OptimizeLevelStyleCompaction`. Since we always set `ROCKSDB_LEVEL_COMPACTION_DYNAMIC_LEVEL_BYTES` to true in clusters, it might be the right time to remove this knob.